### PR TITLE
02 hit the api

### PR DESCRIPTION
**Pull request by gak92**
In this pull request:

- Add functionality to add score into leaderboard using API
- Add functionality to get score list, when click on the refresh button using API
- Use async await to fetch data from API
- Use arrow functions

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,58 @@
...
+};
+
+const getScores = async () => {
+  await fetch(requestedURL)
+    .then((response) => response.json())
+    .then((json) => {
+      renderScores(json.result);
+    });
+};
+
```
Comment (BenMukebo) [inline src/index.js]: - [ ] Great job so far 👏🏻 👏🏻  However According to the project requirement,  you should Use `async` and `await` with the `fetch` method to send and receive data. You're on the good way but you only need to improve you syntax
- Syntax: Follow this syntax which we could use in order to execute an asynchronous method async-await:

```
const getScores = async () => {
  try{
    let response = await fetch(requestedURL);
        let data = await response.json();
        renderScores(data.result);
  } catch(error){
    console.log(error.message)
  }
};

```
- Replace your code with this one 🙃 below, I've already tested 👍🏻 
- For more documentation you can check  [Promises vs Async Await](https://www.loginradius.com/blog/engineering/callback-vs-promises-vs-async-await/)

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @gak92,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- App runs as expected. I can submit user data and it reflects on the scoreboard 👍🏻 

 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (gak92) [reply src/index.js]: @BenMukebo 
I followed this documentation:
https://github.com/microverseinc/curriculum-javascript/blob/main/leaderboard/articles/how_to_consume_api.md
which is mentioned here:
https://github.com/microverseinc/curriculum-javascript/blob/main/leaderboard/lessons/working_with_apis.md

They did not mention using try-catch. I will do the changes as you suggested but I am a little confused. I checked with my other coding partners and all followed the same instructions as mentioned.

If I used console.log then I got linters error -> "using console"
If I don't use then it again gives me linters error -> "empty block"

What should I do?

I appreciate your help and feedback.

Comment (BenMukebo) [reply src/index.js]: Hello @gak92, You don't necessarily need to use `try and catch` it was just an example of how your code should look like. The try statement allows you to define a block of code to be tested for errors while it is being executed. The catch statement allows you to define a block of code to be executed if an error occurs in the try block. if you don't want to use try and catch your code might look like this:


```
  const getScores = async () => {
     let response = await fetch(requestedURL);
        let data = await response.json();
        renderScores(data.result);
  } 
```
- For more documentation you can check [JavaScript try and catch](https://www.w3schools.com/js/js_errors.asp)
- Hope this is more clear now 👍🏻

Comment (gak92) [reply src/index.js]: @BenMukebo 
Thanks
I updated my code as suggested.

**Review by prolajumokeoni — APPROVED** [review]
Hi @gak92, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**